lib/src/test/java/io/ably/lib/test/rest/RestRequestTest.java

```diff
@@ -0,0 +1,287 @@
+		try {
+			Param[] params = new Param[] { new Param("prefix", channelNamePrefix) };
+			PaginatedResult<JsonElement> channels = ably.paginatedRequest(Http.GET, channelsPath, params, null, null);
+			/* check it looks like an array of ChannelDetails */
+			assertNotNull("Verify a result is returned", channels);
+			JsonElement[] items = channels.items();
+			assertTrue("Verify at least two channels are returned", items.length >= 2);
+			for(int i = 0; i < items.length; i++) {
+				assertTrue("Verify id member is a matching channelName", items[i].getAsJsonObject().get("id").getAsString().startsWith(channelNamePrefix));
+			}
+		} catch(AblyException e) {
+			e.printStackTrace();
+			fail("request_simple: Unexpected exception");
+			return;
+		}
+	}
+
+	/**
+	 * Get channel details using the paginatedRequestAsync() API
+	 */
+	@Test
+	public void request_paginated_async() {
+		Param[] params = new Param[] { new Param("prefix", channelNamePrefix) };
+		ably.paginatedRequestAsync(Http.GET, channelsPath, params, null, null, new Callback<AsyncPaginatedResult<JsonElement>>() {
+			@Override
+			public void onSuccess(AsyncPaginatedResult<JsonElement> result) {
+				/* check it looks like an array of ChannelDetails */
+				assertNotNull("Verify a result is returned", result);
+				JsonElement[] items = result.items();
+				assertTrue("Verify at least two channels are returned", items.length >= 2);
+				for(int i = 0; i < items.length; i++) {
+					assertTrue("Verify id member is a matching channelName", items[i].getAsJsonObject().get("id").getAsString().startsWith(channelNamePrefix));
+				}
+			}
+			@Override
+			public void onError(ErrorInfo reason) {
+				fail("request_paginated_async: Unexpected exception");
+			}
+		});
+	}
+
+	/**
+	 * Get channel details using the paginatedRequest() API with a specified limit,
+	 * checking pagination links
+	 */
+	@Test
+	public void request_paginated_limit() {
+		try {
+			Param[] params = new Param[] { new Param("prefix", channelNamePrefix), new Param("limit", "1") };
+			PaginatedResult<JsonElement> channels = ably.paginatedRequest(Http.GET, channelsPath, params, null, null);
+			/* check it looks like an array of ChannelDetails */
+			assertNotNull("Verify a result is returned", channels);
+			JsonElement[] items = channels.items();
+			assertTrue("Verify one channel is returned", items.length == 1);
+			for(int i = 0; i < items.length; i++) {
+				assertTrue("Verify id member is a matching channelName", items[i].getAsJsonObject().get("id").getAsString().startsWith(channelNamePrefix));
+			}
+			/* get next page */
+			channels = channels.next();
+			assertNotNull("Verify a result is returned", channels);
+			items = channels.items();
+			assertTrue("Verify one channels is returned", items.length == 1);
+			for(int i = 0; i < items.length; i++) {
+				assertTrue("Verify id member is a matching channelName", items[i].getAsJsonObject().get("id").getAsString().startsWith(channelNamePrefix));
+			}
```

Comment:
Is it worth checking there are no more pages to at least know that the complete end-to-end paging works?